pyproject.toml

```diff
@@ -62,6 +62,7 @@ register = "twine.commands.register:main"
 
 [project.optional-dependencies]
 keyring = ["keyring >= 15.1"]
+trusted-publishing = ["id"]
```

Comment:
I _think_ it makes sense to move this into the default deps: it's lightweight, and per https://github.com/pypa/twine/issues/999#issuecomment-2517697698 there's a convenient/user-friendly default UX we can enable.

Reply:
`id` itself is lightweight, but it depends on pydantic, the core of which is compiled, so heavy-ish (2 MB wheel). None of twine's dependencies currently use pydantic. I'd maybe still lean towards leaving that out of the default dependencies, but that might just be me. Twine already has more dependencies than I'd probably use.

Reply:
Ah, I forgot about the `pydantic` dep. We only use it for a single class which is itself trivial, so I can look at removing it entirely from `id`.